server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/MailboxSet.scala

```diff
@@ -34,11 +34,12 @@ case class MailboxSetRequest(accountId: AccountId,
                              ifInState: Option[State],
                              create: Option[Map[MailboxCreationId, JsObject]],
                              update: Option[Map[MailboxId, MailboxPatchObject]],
-                             destroy: Option[Seq[MailboxId]],
+                             destroy: Option[Seq[UnparsedMailboxId]],
                              onDestroyRemoveEmails: Option[RemoveEmailsOnDestroy])
 
 object MailboxSetRequest {
   type MailboxCreationId = String Refined NonEmpty
+  type UnparsedMailboxId = String Refined NonEmpty
```

Comment:
As `MailboxCreationId` and `UnparsedMailboxId` are exactly the same type, wouldn't it be better to just keep one generic type here?

Reply:
It is two different usage that happens to share the same constraints yet it is two distinct things.

MailboxCreationId is any non empty string specified by the client

UnparsedMailboxId is # + MailboxCreationId OR a "real" mailboxId

Reply:
yet the type doesn't encode that knowledge.
You could express that with a sum type (an ADT or an Either)